guides/msearch.md

```diff
@@ -0,0 +1,63 @@
+# Multi-Search (msearch)
+
+OpenSearch's Multi-Search (`msearch`) API is a tool that allows you to execute multiple search operations for documents within an index efficiently. This API supports various parameters that provide flexibility in customizing your search operations. It's crucial to understand the key differences between `msearch` and a regular `search` operation in OpenSearch. Unlike a single search operation, `msearch` enables you to perform multiple searches in a single API call, significantly reducing network overhead as it need not initialize a new network connection for each search. This proves particularly advantageous for batch operations or scenarios where searches span multiple indices. By combining searches into one request, `msearch` streamlines the process and enhances overall efficiency. Explore the functionalities of the msearch API with the provided sample code in [../samples/msearch.js](../samples/msearch.js).
```

Comment:
"is a tool that allows you to" -> "allows you to", it's an  API, not a tool :)

I'd be curious to hear what the downsides of msearch are. For example, does msearch require the caller to examine errors themselves?

Reply:
> I'd be curious to hear what the downsides of msearch are. For example, does msearch require the caller to examine errors themselves?

@dblock did you mean to say this could be one of the disadvantages? I will include a couple of cons related to `msearch` in comparison to `search`

Reply:
Yep

Reply:
I have added a con I came across when reading about the difference in both.

> does msearch require the caller to examine errors themselves?

For the above point, I wasn't able to find anything substantial. It would be great if you could link doc/resources so I can add it to the guide.